programs/stake/src/stake_instruction.rs

```diff
@@ -484,17 +485,59 @@ pub fn process_instruction(
             &from_keyed_account::<Rent>(next_keyed_account(keyed_accounts)?)?,
         ),
         StakeInstruction::Authorize(authorized_pubkey, stake_authorize) => {
-            me.authorize(&signers, &authorized_pubkey, stake_authorize)
+            let require_custodian_for_locked_stake_authorize = invoke_context.is_feature_active(
+                &feature_set::require_custodian_for_locked_stake_authorize::id(),
+            );
+
+            if require_custodian_for_locked_stake_authorize {
+                me.authorize(
+                    &signers,
+                    &authorized_pubkey,
+                    stake_authorize,
+                    require_custodian_for_locked_stake_authorize,
+                    &from_keyed_account::<Clock>(next_keyed_account(keyed_accounts)?)?,
+                    keyed_accounts.next().map(|ka| ka.unsigned_key()),
```

Comment:
Is this account ordering correct? I think you're pulling account index 2 here, ie. the next keyed account after Clock, which is the stake/withdraw authority as per docs https://github.com/solana-labs/solana/blob/master/programs/stake/src/stake_instruction.rs#L70, and not necessarily the same as custodian, right?

It might be clearer to include a block naming `next_keyed_account`s, instead of using them inline.

Reply:
Nice catch!

Reply:
Thanks!  The  latest commit addresses this